plugins/by-name/nui/default.nix

```diff
@@ -0,0 +1,11 @@
+{ lib, ... }:
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "nui";
+  originalName = "nui.nvim";
+  package = "nui-nvim";
+  url = "https://github.com/MunifTanjim/nui.nvim/";
+  description = "UI Component Library for Neovim";
+  maintainers = [ lib.maintainers.DataHearth ];
+
+  callSetup = false;
```

Comment:
If this is a plugin with no settings, we probably also want `hasSettings = false`.

Otherwise, we need to "use" `cfg.settings` somewhere.

Reply:
I'm not quite sure. I doesn't have any "main" settings like some other plugins, but work with submodules instead for its configuration. So I would say no for this use-case. I'll update the hasSettings params.

Reply:
Looks like it's got components to be referenced to construct a UI and doesn't have its own settings to configure. Not sure what a user is going to do with this module in the context of nixvim, unless it's just for an easy way to install the plugin through a plugins.*.enable syntax.

Reply:
Well with the addition of plugins.*.luaConfig it can also help segment the configuration

Reply:
Just to play _devil's advocate_ a little:

If the plugin is only really ever used as a dependency of some other plugin, and never really needs to be installed/used by end-users, then we shouldn't really _need_ a `plugins.*.enable` and instead can rely on:

1. Declaring plugin packages depend on this plugin upstream in nixpkgs
2. Adding the plugin-package to nixvim's `extraPlugins` if for whatever reason that isn't done

Reply:
Note that nixvim (or more precisly nixpgks) automatically installs depedent plugins